modules/remotecontrol/index.js

```diff
@@ -64,136 +61,148 @@ class RemoteControl {
-     * is started.
+     * Handles remote control start messages.
+     *
+     * @param {number} id - the id of the request that will be used for the
+     * response.
+     * @param {string} sourceId - The source id of the desktop sharing stream.
      */
-    handlePermissionRequest(userInfo) {
-        this.windowManager.requestRemoteControlPermissions(userInfo)
-            .then(result => {
-                this.sendEvent({
-                    type: EVENT_TYPES.permissions,
-                    action: result ? PERMISSIONS_ACTIONS.grant
-                        : PERMISSIONS_ACTIONS.deny,
-                    userId: userInfo.userId
-                });
-                if(result) {
-                    this.start();
+    start(id, sourceId) {
+        const displays = electron.screen.getAllDisplays();
+
+        switch(displays.length) {
+            case 0:
+                this.display = undefined;
+            break;
+            case 1: // Linux
+                this.display = displays[0];
+            break;
+            // eslint-disable-next-line no-case-declarations
+            default: // > 1 display
+                const coordinates = sourceId2Coordinates(sourceId);
+                if(coordinates) { // Windows
+                    const { x, y } = coordinates;
+                    this.display = electron.screen.getDisplayNearestPoint({
+                        x: x + 1,
+                        y: y + 1
+                    });
+                } else { // ac OS
+                    // try to match displayID with sourceID
+                    const displayID = Number(sourceId.replace('screen:', ''));
+                    this.display
+                        = displays.find(display => display.id === displayID);
                 }
-            }).catch(e => {
-                console.error(e);
-            });
-    }
+        }
 
-    /**
-     * Starts processing the events.
-     */
-    start() {
-        this.started = true;
+        const response = {
+            id,
+            type: 'response'
+        };
+
+        if(this.display) {
+            this.started = true;
+            response.result = true;
+        } else {
+            response.error
+                = 'Error: Can\'t detect the display that is currently shared';
+        }
+
+        this.sendResponse(response);
     }
 
     /**
      * Stops processing the events.
      */
     stop() {
         this.started = false;
+        this.display = undefined;
     }
 
     /**
-     * Executes the passed event.
-     * @param {Object} event the remote-control-event.
+     * Executes the passed message.
+     * @param {Object} message the remote-control message.
      */
-    onRemoteControlEvent(event) {
-        if(!this.started && event.type !== EVENT_TYPES.permissions) {
+    onRemoteControlMessage(message) {
+        const { id, data } = message;
+
+        if(!this.started
+            && data.type != REQUESTS.start) {
```

Comment:
This is the remote control message format coming from meet. You can check modules/transport in Jitsi Meet project. What happens is that Transport class implements sending requests. The format for the requests is `{id, type: 'request', data}`. `data` prop is defined by every consumer that uses the Transport module. The remote control module in Jitsi Meet project uses this field too. 

All remote control messages have `data.name = 'remote-control'`. This way the electron app can filter them from all other messages. Also all remote control messages are defining `data.type` to specify an operation to be executed by the electron app.

In this case we have `data.type = 'start'` which we are using to notify the electron app that a remote control session will start and to pass some initial data that will be needed( like the desktop sharing source id) by the electron app.

Do you have any suggestions for improvement? Maybe renaming?